workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/SpatialOSBehaviourLibrary.cs

```diff
@@ -9,19 +11,17 @@ namespace Improbable.Gdk.Core
     /// <summary>
     ///     Retrieves Reader and Writer fields from MonoBehaviours and handles injection into them.
     /// </summary>
-    public class SpatialOSBehaviourLibrary
+    internal class SpatialOSBehaviourLibrary
     {
-        private readonly Dictionary<Type, Dictionary<uint, FieldInfo>> fieldInfoCache
-            = new Dictionary<Type, Dictionary<uint, FieldInfo>>();
+        private readonly Dictionary<Type, Dictionary<uint, FieldInfo[]>> fieldInfoCache
+            = new Dictionary<Type, Dictionary<uint, FieldInfo[]>>();
 
         private readonly Dictionary<Type, List<uint>> componentReaderIdsForBehaviours =
             new Dictionary<Type, List<uint>>();
 
         private readonly Dictionary<Type, List<uint>> componentWriterIdsForBehaviours =
             new Dictionary<Type, List<uint>>();
 
-        private readonly HashSet<Type> invalidMonoBehaviourTypes = new HashSet<Type>();
```

Comment:
We had previously decided that in that case the invalid required type is simply ignored, logging an error, without affecting anything. (This could easily be changed of course, but seems reasonable to me.)

Reply:
Ok nice, sounds good. :)